engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -2116,9 +2126,25 @@ namespace dmGameSystem
 
     dmGameObject::PropertyResult CompGuiSetProperty(const dmGameObject::ComponentSetPropertyParams& params) {
         GuiComponent* gui_component = (GuiComponent*)*params.m_UserData;
-        if (params.m_PropertyId == PROP_MATERIAL) {
+        dmhash_t set_property = params.m_PropertyId;
+        if (set_property == PROP_MATERIAL) {
             return SetResourceProperty(dmGameObject::GetFactory(params.m_Instance), params.m_Value, MATERIAL_EXT_HASH, (void**)&gui_component->m_Material);
         }
+        else if (set_property == PROP_FONTS) {
+            if (!params.m_Options.m_Key) {
+                return dmGameObject::PROPERTY_RESULT_INVALID_KEY;
+            }
+            dmRender::HFontMap font = 0x0;
+            dmGameObject::PropertyResult res = SetResourceProperty(dmGameObject::GetFactory(params.m_Instance), params.m_Value, FONT_EXT_HASH, (void**)&font);
+            dmGui::HScene scene = gui_component->m_Scene;
+            dmGui::Result r = dmGui::AddFont(scene, params.m_Options.m_Key, (void*) font, params.m_Value.m_Hash);
+            gui_component->m_resourcePropertyPointers.Push(font);
+            if (r != dmGui::RESULT_OK) {
+                dmLogError("Unable to set font property");
+                return dmGameObject::PROPERTY_RESULT_BUFFER_OVERFLOW;
```

Comment:
Don't we want to decrement the reference here?
I.e. I don't think we should keep the resource if setting it failed

Reply:
Good point!